src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/code_client.py

```diff
@@ -23,16 +23,16 @@ def __init__(self, run, input_data, evaluator_name=None, aggregated_metrics=None
         self.input_data = input_data
         self.aggregated_metrics = aggregated_metrics
 
-    def get_result_df(self, exclude_inputs=False):
-        result_df = self.run.result(timeout=BATCH_RUN_TIMEOUT)
+    def get_result_df(self, exclude_inputs=False, batch_run_timeout=BATCH_RUN_TIMEOUT):
+        result_df = self.run.result(timeout=batch_run_timeout)
         if exclude_inputs:
             result_df = result_df.drop(columns=[col for col in result_df.columns if col.startswith("inputs.")])
         return result_df
 
-    def get_aggregated_metrics(self):
+    def get_aggregated_metrics(self, batch_run_timeout=BATCH_RUN_TIMEOUT):
```

Comment:
Can users control the timeout ?

Reply:
yes, this was just setting the BATCH_RUN_TIMEOUT constant as a default; however, I've now changed this to be read via environment variable rather than a method argument.